### PR TITLE
Cleanup EventInfo tests

### PR DESCRIPTION
**Pull request by hughbe**
/cc @stephentoub

### CODE REVIEW

File: src/System.Reflection/tests/AssemblyTests.cs [hunk 1]
```diff
@@ -110,7 +110,6 @@ public void Equals(Assembly assembly1, Assembly assembly2, bool expected)
         [InlineData(typeof(AssemblyTests), true)]
         [InlineData(typeof(AssemblyPublicClass.PublicNestedClass), true)]
         [InlineData(typeof(PublicEnum), true)]
-        [InlineData(typeof(BaseClass), true)]
```
Comment (hughbe) [inline src/System.Reflection/tests/AssemblyTests.cs]: The type no longer exists - we have enough tests for ExportedTypes so I'm not concerned about this getting removed

File: src/System.Reflection/tests/EventInfo/EventInfo_PropertyTests.cs [hunk 2]
```diff
@@ -1,294 +0,0 @@
...
-
-#pragma warning disable 0067
-
-namespace System.Reflection.Tests
-{
-    public class EventInfoPropertyTests
-    {
-        // Verify AddMethod for EventInfo
-        [Fact]
-        public void VerifyAddMethod1()
```
Comment (hughbe) [inline src/System.Reflection/tests/EventInfo/EventInfo_PropertyTests.cs]: The tests for the AddMethod, RemoveMethod and RaiseMethod properties are now combined into the tests for GetAddMethod(), GetRemoveMethod() and GetRaiseMethod() respectively

File: src/System.Reflection/tests/EventInfo/EventInfo_PropertyTests.cs [hunk 3]
```diff
@@ -1,294 +0,0 @@
...
-            EventInfo ei = GetEventInfo(typeof(EventInfoPropertyBaseClass), "EventPublicVirtual");
-            Assert.NotNull(ei);
-            EventAttributes attr = ei.Attributes;
-            EventAttributes expectedAttr = EventAttributes.None;
-            Assert.Equal((object)expectedAttr, (object)attr);
-        }
-
-        //Verify IsSpecialName Property
-        [Fact]
-        public void VerifyIsSpecialName1()
```
Comment (hughbe) [inline src/System.Reflection/tests/EventInfo/EventInfo_PropertyTests.cs]: These tests have been inlined into the tests for the Attributes property

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Reflection/tests/EventInfoTests.cs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+        [InlineData(typeof(BaseClass), "PrivateEvent", false)]
+        public void GetAddMethod(Type type, string name, bool isVisible)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            Assert.Equal(!isVisible, eventInfo.GetAddMethod() == null);
+            Assert.Equal(!isVisible, eventInfo.GetAddMethod(false) == null);
+            Assert.NotNull(eventInfo.GetAddMethod(true));
+
+            MethodInfo addMethod = eventInfo.AddMethod;
+            Assert.Equal(eventInfo.GetAddMethod(true), addMethod);
```
Comment (stephentoub) [inline src/System.Reflection/tests/EventInfoTests.cs]: Nit: should these args be swapped?

File: src/System.Reflection/tests/EventInfoTests.cs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        [InlineData(typeof(BaseClass), "PrivateEvent", false)]
+        public void GetRemoveMethod(Type type, string name, bool isVisible)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            Assert.Equal(!isVisible, eventInfo.GetRemoveMethod() == null);
+            Assert.Equal(!isVisible, eventInfo.GetRemoveMethod(false) == null);
+            Assert.NotNull(eventInfo.GetRemoveMethod(true));
+
+            MethodInfo removeMethod = eventInfo.RemoveMethod;
+            Assert.Equal(eventInfo.GetRemoveMethod(true), removeMethod);
```
Comment (stephentoub) [inline src/System.Reflection/tests/EventInfoTests.cs]: Same nit

File: src/System.Reflection/tests/EventInfoTests.cs [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+            Assert.Equal(typeof(EventHandler), eventInfo.EventHandlerType);
+        }
+
+        [Theory]
+        [MemberData(nameof(Events_TestData))]
+        public void Attributes(Type type, string name)
+        {
+            EventInfo eventInfo = GetEventInfo(type, name);
+            Assert.Equal(EventAttributes.None, eventInfo.Attributes);
+            Assert.False(eventInfo.IsSpecialName);
```
Comment (stephentoub) [inline src/System.Reflection/tests/EventInfoTests.cs]: Seems like this IsSpecialName check should be in its own test.  Is there a connection with Attributes I'm not seeing?

**Comment by stephentoub** [comment]
A few comments, otherwise LGTM. Thanks, @hughbe.

File: src/System.Reflection/tests/EventInfoTests.cs (hunk 6 above)
Comment (hughbe) [inline src/System.Reflection/tests/EventInfoTests.cs]: Well, the code in coreclr is:
```
        public bool IsSpecialName 
        {
            get 
            {
                return(Attributes & EventAttributes.SpecialName) != 0;
            }
        }
```

So I saw an implicit connection between Attributes and IsSpecialName, so they could be tested together - let me know if you'd still like me to change this

Comment (stephentoub) [inline src/System.Reflection/tests/EventInfoTests.cs]: Hmm, fair enough, but I think that's an implementation detail we shouldn't reflect in the tests.  Ideally I think it'd be separated out.

Comment (hughbe) [inline src/System.Reflection/tests/EventInfoTests.cs]: Sure - I'll separate it out when I address the other PR feedback

File: src/System.Reflection/tests/EventInfoTests.cs (hunk 4 above)
Comment (hughbe) [inline src/System.Reflection/tests/EventInfoTests.cs]: fixed

File: src/System.Reflection/tests/EventInfoTests.cs (hunk 5 above)
Comment (hughbe) [inline src/System.Reflection/tests/EventInfoTests.cs]: fixed, thanks

File: src/System.Reflection/tests/EventInfoTests.cs (hunk 6 above)
Comment (hughbe) [inline src/System.Reflection/tests/EventInfoTests.cs]: I've split this up - thansk

**Pull request merged**